lib/sales_engine.rb

```diff
@@ -99,11 +99,55 @@ def customer_merchants(id)
   end
 
   def total(id)
+  return 0 if !invoices.find_by_id(id).is_paid_in_full?
   selected_invoices = invoice_items.find_all_by_invoice_id(id)
   totals_of_invoice_items = selected_invoices.map do |invoice_item|
     (invoice_item.quantity * invoice_item.unit_price)
     end
     totals_of_invoice_items.reduce(:+)
   end
 
+  def merchants_by_total_revenue
+    merchants.merchants_by_total_revenue
+  end
+
+  def revenue_by_merchant(merchant_id)
+    merchants.revenue_by_merchant(merchant_id)
+  end
+
+  def most_sold_item_for_merchant(merchant_id)
+    invoice_list = invoices.find_all_by_merchant_id(merchant_id)
+    paid_invoices = invoice_list.select { |inv| inv.is_paid_in_full? }
+    iis = paid_invoices.map { |ii| ii.invoice_items }
+    turn_to_items(top_item(iis))
+  end
+
+  def top_item(iis)
+    max = iis.flatten.max_by { |ii| ii.quantity }
+    iis.flatten.select do |ii|
+      ii.quantity == max.quantity
+    end
+  end
```

Comment:
@erikaannesmith @Tyjoo26 
What your variable names here. iis is a bit difficult to track what they refer to.